Dockerfile

```
@@ -28,17 +28,46 @@ ENV NODE_ENV=production
-    /app/node_modules/youtube-dl-exec/bin/yt-dlp \
-    ./node_modules/youtube-dl-exec/bin/yt-dlp
+# ── yt-dlp ───────────────────────────────────────────────────────────────────
+# Install via pip3 (pure Python — architecture-independent, works on Alpine
+# musl without glibc issues). Then expose it at the path selectYtDlpPath()
+# probes: node_modules/youtube-dl-exec/bin/yt-dlp.
+#
+# PyPI strips leading zeros from version components:
+#   .ytdlp-version: 2026.03.17  →  pip version: 2026.3.17
+# awk handles the conversion.
+COPY .ytdlp-version .ytdlp-version
+RUN YTDLP_VERSION=$(awk -F. '{printf "%d.%d.%d",$1,$2,$3}' .ytdlp-version) && \
```

Hadolint (2.14.0)

[warning] Avoid use of cache directory with pip. Use `pip install --no-cache-dir <package>`

(DL3042)